File: leetcode_py/Remove Nth Node From End of List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @return a ListNode
    def removeNthFromEnd(self, head, n):
        if n == 0:
            return head
        dummy = ListNode(0)
        dummy.next = head
        pre = dummy
        nextn = dummy
        for i in range(n):
            nextn = nextn.next
        while nextn.next:
            pre = pre.next
            nextn = nextn.next
        pre.next = pre.next.next
        return dummy.next

File: leetcode_js/Remove Nth Node From End of List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
	if(n == 0) return head;
	var dummy = new ListNode(0);
	dummy.next = head;
    var ahead = cur = dummy;
    while(n-- >0){ahead = ahead.next;}
    while(ahead.next != null){
    	ahead = ahead.next;
    	cur = cur.next;
    }
    cur.next = cur.next.next;
    return dummy.next;
};